controller/consul.go

```diff
@@ -97,27 +98,73 @@ func (c *ConsulMon) queryServices() ([]*App, error) {
 	return apps, nil
 }
 
-func (c *ConsulMon) healthCheck(service string) (bool, error) {
-	addr := c.addr + fmt.Sprintf("%s/%s", healthCheckurl, service)
+func getResp(addr) (resp Response) {
 	resp, err := http.Get(addr)
 	if err != nil {
-		return false, err
+		glog.V(2).Errorf("Error getting %s with %s", addr, err)
 	}
 	defer resp.Body.Close()
+	return resp
+}
+
+// healthCheckLocal queries a node's local consul agent to perform service healthchecks
+// This is the underlying api call: https://www.consul.io/api/agent/check.html
+func (c *ConsulMon) healthCheckLocal(service string) (bool, error) {
+	addr := c.addr + fmt.Sprintf("%s/%s", localHealthCheckurl, service)
+	resp, err = getResp(addr)
+
+	var data interface{}
+	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
+		return false, err
+	}
+
+	services := data.(map[string]interface{})
+	for _, serviceInfo := range services {
+		s := serviceInfo.(map[string]interface{})
+		serviceName := s["ServiceName"].(string)
+		if serviceName == service {
+			status := s["Status"].(string)
+			if status == "passing" {
+				return true, nil
+			}
+			glog.V(2).Infof("Consul local healthcheck returned %s status", status)
+			return false, nil
+		}
+		node := serviceReport["Node"].(string)
+		return false, fmt.Errorf("No local healcheck info found for service %s on node %s in consul", service, node)
+	}
+}
+
+// healthCheckRemote queries the consul cluster's healthcheck endpoint to perform service healthchecks
+// This is the underlying api call: https://www.consul.io/api/health.html
+func (c *ConsulMon) healthCheckRemote(service string) (bool, error) {
+	addr := c.addr + fmt.Sprintf("%s/%s", remoteHealthCheckurl, service)
+	resp, err = getResp(addr)
+
 	var data []interface{}
 	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
 		return false, err
 	}
+
 	for _, nodeInfo := range data {
 		n := nodeInfo.(map[string]interface{})
-		if n["Node"] == c.node {
+		if n["Node"].(string) == c.node {
```

Comment:
It's possible this didn't work originally, so I added a type assertion to be sure.